platform/pulpcore/app/viewsets/task.py

```diff
@@ -19,18 +20,19 @@ class Meta:
         fields = ['state', 'worker__name', 'tags']
 
 
-class TaskViewSet(NamedModelViewSet):
+class TaskViewSet(mixins.RetrieveModelMixin,
+                  mixins.ListModelMixin,
+                  GenericNamedModelViewSet):
     queryset = Task.objects.all()
     serializer_class = TaskSerializer
     endpoint_name = 'tasks'
     filter_class = TaskFilter
-    http_method_names = ['get', 'options']
 
-    @detail_route(methods=['post'])
+    @detail_route(methods=('post',))
```

Comment:
Why does it fix the issue? Is it because by not including "create" mixin there is no default handler for POST but POST itself is not explicitly disallowed and you can specify it in detail route?

Reply:
This line actually doesn't change anything, I just switched to a tuple to be consistent with other parts of the code.

Reply:
Yeah, I just meant that you can specify POST while it seems to be disallowed for the viewset as a whole. But I think my understanding is correct that it's not actually disallowed but default "create" method is just not defined. Never mind :) Thanks